xray/plot/facetgrid.py

```diff
@@ -0,0 +1,383 @@
+        rep_row = row is not None and not darray[row].to_index().is_unique
+        if rep_col or rep_row:
+            raise ValueError('Coordinates used for faceting cannot '
+                             'contain repeated (nonunique) values.')
+
+        # self._single_group is the grouping variable, if there is exactly one
+        if col and row:
+            self._single_group = False
+            self._nrow = len(darray[row])
+            self._ncol = len(darray[col])
+            self.nfacet = self._nrow * self._ncol
+            if col_wrap is not None:
+                warnings.warn('Ignoring col_wrap since both col and row '
+                              'were passed')
+        elif row and not col:
+            self._single_group = row
+        elif not row and col:
+            self._single_group = col
+        else:
+            raise ValueError(
+                'Pass a coordinate name as an argument for row or col')
+
+        # Compute grid shape
+        if self._single_group:
+            self.nfacet = len(darray[self._single_group])
+            if col:
+                # idea - could add heuristic for nice shapes like 3x4
+                self._ncol = self.nfacet
+            if row:
+                self._ncol = 1
+            if col_wrap is not None:
+                # Overrides previous settings
+                self._ncol = col_wrap
+            self._nrow = int(np.ceil(self.nfacet / self._ncol))
+
+        # Calculate the base figure size with extra horizontal space for a
+        # colorbar
+        self._cbar_space = 1
+        figsize = (self._ncol * size * aspect +
+                   self._cbar_space, self._nrow * size)
+
+        self.fig, self.axes = plt.subplots(self._nrow, self._ncol,
+                                           sharex=True, sharey=True,
+                                           squeeze=False, figsize=figsize)
+
+        # Set up the lists of names for the row and column facet variables
+        col_names = list(darray[col].values) if col else []
+        row_names = list(darray[row].values) if row else []
+
+        if self._single_group:
+            full = [{self._single_group: x} for x in
+                    darray[self._single_group].values]
+            empty = [None for x in range(self._nrow * self._ncol - len(full))]
+            name_dicts = full + empty
+        else:
+            rowcols = itertools.product(row_names, col_names)
+            name_dicts = [{row: r, col: c} for r, c in rowcols]
+
+        self.name_dicts = np.array(name_dicts).reshape(self._nrow, self._ncol)
+
+        self.row_names = row_names
+        self.col_names = col_names
+        self.darray = darray
+        self.row = row
+        self.col = col
+        self.col_wrap = col_wrap
+
+        # Next the private variables
+        self._row_var = row
+        self._col_var = col
+        self._col_wrap = col_wrap
+
+        self.set_titles()
+
+    def __iter__(self):
```

Comment:
My inclination is to not add this convenience shortcut, for two reasons:

1. It's a departure from Seaborn, so it's something new for users to learn.
2. It's not really necessary -- iterating over `grid.axes.flat` is much more explicit (generally a good thing), and only slightly more verbose.